.github/workflows/main.yaml

```diff
@@ -29,16 +29,16 @@ jobs:
           # on m1 the R version is whicherver is installed in the runner machine.
           - {os: macOS, r_version: '', version: cpu-m1, runner: [self-hosted, macOS, ARM64]}
 
-          - {os: ubuntu, r_version: release, version: cpu, runner: ubuntu-20.04}
-          - {os: ubuntu, r_version: release, version: cpu, runner: ubuntu-20.04, precxx11abi: 1}
+          - {os: ubuntu, r_version: release, version: cpu, runner: ubuntu-22.04}
+          - {os: ubuntu, r_version: release, version: cpu, runner: ubuntu-22.04, precxx11abi: 1}
           - {os: ubuntu, r_version: release, version: cu12.4, runner: [self-hosted, gpu-local]}
 
           - {os: windows, r_version: release, version: cpu, runner: windows-latest}
         
         include:
           
           - config: {os: ubuntu, version: cu12.4}
-            container: {image: 'nvidia/cuda:12.4.1-cudnn-devel-ubuntu20.04', options: '--gpus all --runtime=nvidia'}
+            container: {image: 'nvidia/cuda:12.4.1-cudnn-devel-ubuntu22.04', options: '--gpus all --runtime=nvidia'}
```

Comment:
Let's not update the containers, as its better to build on older versions for maximum compatibility.